tests/test_bootstrap.py

```diff
@@ -0,0 +1,1390 @@
+
+        # Set initial condition:
+        Rmajor = 6.0
+        aminor = 2.0
+        NFP = 1
+        surface = FourierRZToroidalSurface(
+            R_lmn=np.array([Rmajor, aminor]),
+            modes_R=[[0, 0], [1, 0]],
+            Z_lmn=np.array([-aminor]),
+            modes_Z=[[-1, 0]],
+            NFP=NFP,
+        )
+
+        eq = Equilibrium(
+            surface=surface,
+            electron_density=ne,
+            electron_temperature=Te,
+            ion_temperature=Ti,
+            atomic_number=Zeff,
+            iota=iota,
+            Psi=B0 * np.pi * (aminor**2),
+            NFP=NFP,
+            L=LM_resolution,
+            M=LM_resolution,
+            N=0,
+            L_grid=2 * LM_resolution,
+            M_grid=2 * LM_resolution,
+            N_grid=0,
+            sym=True,
+        )
+
+        eq.solve(
+            verbose=3,
+            ftol=1e-8,
+            constraints=get_fixed_boundary_constraints(kinetic=True),
+            optimizer=Optimizer("lsq-exact"),
+            objective=ObjectiveFunction(objectives=ForceBalance()),
+        )
+
+        eq.save("test_bootstrap_consistency_iota_initial.h5")
+        # eq = desc.io.load("test_bootstrap_consistency_iota_initial.h5")
+
+        # Done establishing the initial condition. Now set up the optimization.
+
+        constraints = (
+            ForceBalance(),
+            FixBoundaryR(),
+            FixBoundaryZ(),
+            FixElectronDensity(),
+            FixElectronTemperature(),
+            FixIonTemperature(),
+            FixAtomicNumber(),
+            FixPsi(),
+        )
+
+        # grid for bootstrap consistency objective:
+        grid = LinearGrid(
+            rho=np.linspace(1e-4, 1 - 1e-4, (num_iota_points - 1) * 2 + 1),
+            M=eq.M * 2,
+            N=eq.N * 2,
+            NFP=eq.NFP,
+        )
+        objective = ObjectiveFunction(
+            BootstrapRedlConsistency(
+                grid=grid,
+                helicity=helicity,
+            )
+        )
+        eq, _ = eq.optimize(
+            verbose=3,
+            objective=objective,
+            constraints=constraints,
+            optimizer=Optimizer("scipy-trf"),
+            ftol=1e-6,
+        )
+
+        eq.save("test_bootstrap_consistency_iota_final.h5")
+
+        scalar_objective = objective.compute_scalar(objective.x(eq))
+        assert scalar_objective < 3e-5
+        data = eq.compute(
+            ["<J*B>", "<J*B> Redl"],
+            grid=grid,
+            helicity=helicity,
+        )
+        J_dot_B_MHD = compress(grid, data["<J*B>"])
+        J_dot_B_Redl = compress(grid, data["<J*B> Redl"])
+
+        assert np.max(J_dot_B_MHD) < 4e5
+        assert np.max(J_dot_B_MHD) > 0
+        assert np.min(J_dot_B_MHD) < -5.1e6
+        assert np.min(J_dot_B_MHD) > -5.4e6
+        np.testing.assert_allclose(J_dot_B_MHD, J_dot_B_Redl, atol=5e5)
+
+    # MJL: Should this test be decorated with @pytest.mark.slow ?
+    @pytest.mark.unit
```

Comment:
same note about making this a regression test

Reply:
This has been changed to `regression`.